CMakeLists.txt

```diff
@@ -14,19 +15,56 @@ IF(UNIX)
         INCLUDE(GNUInstallDirs)
     ENDIF()
 ENDIF()
-SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fwrapv -DNDEBUG")
 
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fwrapv -DNDEBUG -DUSE_GLOP -DUSE_BOP")
 SET(BUILD_CXX ON)
+SET(DOWNLOAD_LOCATION "${CMAKE_CURRENT_BINARY_DIR}/downloads"
+        CACHE PATH "Location where external projects will be downloaded.")
+
+MARK_AS_ADVANCED(BUILD_CXX)
+MARK_AS_ADVANCED(DOWNLOAD_LOCATION)
+
+SET(${PROJECT_NAME}externalLibraries "")
+SET(${PROJECT_NAME}externalTargets "")
 
 IF(BUILD_CXX)
-    SET(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -DUSE_GLOP -DUSE_BOP")
+    FIND_PACKAGE(Threads REQUIRED)
 
-    FIND_PACKAGE(Protobuf REQUIRED)
+    SET(Protobuf_VERSION 3.0.2)
```

Comment:
It is! However, I've kept it the same as the one in dependencies folder. Ideal would be to pick all of these up from a command config file that both current make and CMake can use and you could bump at will!